lib/galaxy/tools/error_reports/plugins/github.py

```diff
@@ -20,73 +26,126 @@ def __init__(self, **kwargs):
+            github_projecturl = urlparse.urlparse(ts_repourl).path[1:] if ts_repourl else \
+                "/".join([self.git_default_repo_owner, self.git_default_repo_name])
+            issue_cache_key = self._get_issue_cache_key(job, ts_repourl)
 
-            # The HTML report
-            error_message = error_reporter.html_report
+            # Connect to the repo
+            if github_projecturl not in self.git_project_cache:
+                self.git_project_cache[github_projecturl] = self.github.get_repo('%s' % github_projecturl)
+            gh_project = self.git_project_cache[github_projecturl]
 
-            log.info(error_title in self.issue_cache)
-            if error_title not in self.issue_cache:
+            # Make sure we keep a cache of the issues, per tool in this case
+            if issue_cache_key not in self.issue_cache:
+                self._fill_issue_cache(gh_project, issue_cache_key)
+
+            # Retrieve label
+            label = self.get_label('%s/%s' % (unicodify(job.tool_id), unicodify(job.tool_version)), gh_project, issue_cache_key)
+
+            # Generate information for the tool
+            error_title = self._generate_error_title(job)
+
+            # Generate the error message
+            error_message = self._generate_error_message(dataset, job, kwargs)
+
+            log.info(error_title in self.issue_cache[issue_cache_key])
+            if error_title not in self.issue_cache[issue_cache_key]:
                 # Create a new issue.
-                self.issue_cache[error_title] = self.repo.create_issue(
-                    title=error_title,
-                    body=error_message,
-                    # Label it with a tag: tool_id/tool_version
-                    labels=[label]
-                )
+                self._create_issue(error_message=error_message, error_title=error_title, gh_project=gh_project,
+                                   issue_cache_key=issue_cache_key, label=label)
             else:
-                self.issue_cache[error_title].create_comment(error_message)
-            return ('Submitted bug report to Github. Your issue number is %s' % self.issue_cache[error_title].number, 'success')
+                self._append_issue(issue_cache_key=issue_cache_key, error_title=error_title, error_message=error_message)
+            return ('Submitted error report to Github. Your issue number is <a href="%s/%s/issues/%s" '
+                    'target="_blank">#%s</a>.' % (self.github_base_url, github_projecturl,
+                                                  self.issue_cache[issue_cache_key][error_title].number,
+                                                  self.issue_cache[issue_cache_key][error_title].number), 'success')
+
+    def _create_issue(self, **kwargs):
+        # Create a new issue.
+        self.issue_cache[kwargs.get('issue_cache_key')][kwargs.get('error_title')] = kwargs.get('gh_project').create_issue(
```

Comment:
Would you mind just make these explicit function parameters rather than pulling from kwargs, since all of the arguments are required? Same for `_append_issue`